Source/WebCore/loader/archive/cf/LegacyWebArchive.h

```diff
@@ -43,18 +44,26 @@ struct SimpleRange;
 class LegacyWebArchive final : public Archive {
 public:
     WEBCORE_EXPORT static Ref<LegacyWebArchive> create();
+    WEBCORE_EXPORT static Ref<LegacyWebArchive> create(Ref<ArchiveResource>&& mainResource, Vector<Ref<ArchiveResource>>&& subresources, Vector<FrameIdentifier>&& subframeIdentifiers);
     WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(FragmentedSharedBuffer&);
     WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(const URL&, FragmentedSharedBuffer&);
     WEBCORE_EXPORT static Ref<LegacyWebArchive> create(Ref<ArchiveResource>&& mainResource, Vector<Ref<ArchiveResource>>&& subresources, Vector<Ref<LegacyWebArchive>>&& subframeArchives);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(Node&, NOESCAPE const Function<bool(LocalFrame&)>& frameFilter = { }, const Vector<MarkupExclusionRule>& markupExclusionRules = { }, const String& mainFrameFileName = { }, bool saveScriptsFromMemoryCache = true);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(LocalFrame&);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> createFromSelection(LocalFrame*, bool saveScriptsFromMemoryCache = true);
-    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(const SimpleRange&, bool saveScriptsFromMemoryCache = true);
+    enum class ShouldSaveSubframeArchives : bool { No, Yes };
+    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(Node&, NOESCAPE const Function<bool(LocalFrame&)>& frameFilter = { }, const Vector<MarkupExclusionRule>& markupExclusionRules = { }, const String& mainFrameFileName = { }, bool saveScriptsFromMemoryCache = true, ShouldSaveSubframeArchives = ShouldSaveSubframeArchives::Yes);
+    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(LocalFrame&, ShouldSaveSubframeArchives = ShouldSaveSubframeArchives::Yes);
+    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> createFromSelection(LocalFrame*, bool saveScriptsFromMemoryCache = true, ShouldSaveSubframeArchives = ShouldSaveSubframeArchives::Yes);
+    WEBCORE_EXPORT static RefPtr<LegacyWebArchive> create(const SimpleRange&, bool saveScriptsFromMemoryCache = true, ShouldSaveSubframeArchives = ShouldSaveSubframeArchives::Yes);
 
     WEBCORE_EXPORT RetainPtr<CFDataRef> rawDataRepresentation();
 
+    Ref<ArchiveResource> protectedMainResource() const { return *mainResource(); }
+    void setSubframeIdentifiers(Vector<FrameIdentifier>&& identifiers) { m_subframeIdentifiers = WTFMove(identifiers); }
```

Comment:
Given this function is only ever called right after `create` function, I think it would be better to add a new create function which takes the vector as an input.